ebu_tt_live/documents/ebutt3.py

```diff
@@ -717,12 +717,20 @@ def extract_segment(self, begin=None, end=None, sequence_number=None, discard=Fa
         for doc in affected_documents:
             doc_ending = doc.resolved_end_time
             if end is not None:
-                if end < doc_ending:
+                if doc_ending is None or end < doc_ending:
                     doc_ending = end
             # Check only til resolved end, otherwise there will be unwanted parallel elements
-            doc_segment = doc.extract_segment(begin=begin, end=doc_ending, deconflict_ids=True)
-
-            document_segments.append(doc_segment)
+            try:
+                doc_segment = doc.extract_segment(begin=begin, end=doc_ending, deconflict_ids=True)
+                document_segments.append(doc_segment)
+            except Exception as err:
+                log.error(
+                    'Error extracting document segment from {}__{}'.format(
```

Comment:
No it is not bad practice but the problem here is the lack of knowledge of what the cases are at the moment. I did the stacktrace originally to see what the issues can be. There could be numerous ways a segmentation can go wrong and I did not manage to explore the problem space yet.